sg_covid_impact/secondary_data.py

```diff
@@ -0,0 +1,105 @@
+import pandas as pd
+import sg_covid_impact
+import logging
+import numpy as np
+
+project_dir = sg_covid_impact.project_dir
+
+_APS_URL = "https://www.nomisweb.co.uk/api/v01/dataset/NM_17_5.data.csv?geography=1811939329...1811939332,1811939334...1811939336,1811939338...1811939497,1811939499...1811939501,1811939503,1811939505...1811939507,1811939509...1811939517,1811939519,1811939520,1811939524...1811939570,1811939575...1811939599,1811939601...1811939628,1811939630...1811939634,1811939636...1811939647,1811939649,1811939655...1811939664,1811939667...1811939680,1811939682,1811939683,1811939685,1811939687...1811939704,1811939707,1811939708,1811939710,1811939712...1811939717,1811939719,1811939720,1811939722...1811939730&date=latestMINUS2&variable=18,45,290,335,344&measures=20599,21001,21002,21003"
+_ASHE_URL = "https://www.nomisweb.co.uk/api/v01/dataset/NM_30_1.data.csv?geography=1811939329...1811939332,1811939334...1811939336,1811939338...1811939497,1811939499...1811939501,1811939503,1811939505...1811939507,1811939509...1811939517,1811939519,1811939520,1811939524...1811939570,1811939575...1811939599,1811939601...1811939628,1811939630...1811939634,1811939636...1811939647,1811939649,1811939655...1811939664,1811939667...1811939680,1811939682,1811939683,1811939685,1811939687...1811939704,1811939707,1811939708,1811939710,1811939712...1811939717,1811939719,1811939720,1811939722...1811939730&date=latest&sex=8&item=2&pay=7&measures=20100,20701"
+_SIMD_URL = "https://www.gov.scot/binaries/content/documents/govscot/publications/statistics/2020/01/scottish-index-of-multiple-deprivation-2020-ranks-and-domain-ranks/documents/scottish-index-of-multiple-deprivation-2020-ranks-and-domain-ranks/scottish-index-of-multiple-deprivation-2020-ranks-and-domain-ranks/govscot%3Adocument/SIMD%2B2020v2%2B-%2Branks.xlsx"
+_DZ_LU = "http://statistics.gov.scot/downloads/file?id=2a2be2f0-bf5f-4e53-9726-7ef16fa893b7%2FDatazone2011lookup.csv"
+_SECOND = f"{project_dir}/data/processed/lad_secondary.csv"
+
+
+def fetch_process_nomis(
+    url, indicator_name, value_column, source, indicator_column="MEASURES_NAME"
+):
+    """Fetch nomis data
+    Args:
+        url (str): API url
+        indicator_name (str): name of indicator
+        value_column (str): value column
+        source (str): data source
+        indicator_column (str): column that contains the indicator
+    Returns:
+        A clean table with secondary data
+    """
+    logging.info(f"Fetching {source}")
+    df = pd.read_csv(url)
+    df_sel = df.query(f"{indicator_column}=='{indicator_name}'")[
+        ["DATE", "GEOGRAPHY_NAME", "GEOGRAPHY_CODE", value_column, "OBS_VALUE"]
+    ].reset_index(drop=True)
+    df_sel.rename(
+        columns={"OBS_VALUE": "VALUE", value_column: "VARIABLE"}, inplace=True
```

Comment:
`inplace` is considered bad practice because it obscures the fact that `df_sel` is mutated.

Reply:
In fact this whole function may be expressed as a single expression

```
    return (
        pd.read_csv(url)
        .query(f"{indicator_column}=='{indicator_name}'")[
            ["DATE", "GEOGRAPHY_NAME", "GEOGRAPHY_CODE", value_column, "OBS_VALUE"]
        ]
        .reset_index(drop=True)
        .rename(columns={"OBS_VALUE": "VALUE", value_column: "VARIABLE"})
        .assign(source=source)
        .rename(columns=str.lower)
    )
```